linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/IncOptimizer.scala

```diff
@@ -348,15 +350,17 @@ final class IncOptimizer private[optimizer] (config: CommonPhaseConfig, collOps:
     val reverseParentChain: List[Class] =
       parentChain.reverse
 
-    var interfaces: Set[InterfaceType] = Set.empty
+    var interfaces: Set[InterfaceType] = linkedClass.ancestors.map(getInterface).toSet
     var subclasses: collOps.ParIterable[Class] = collOps.emptyParIterable
-    var isInstantiated: Boolean = false
+    var isInstantiated: Boolean = linkedClass.hasInstances
 
     var hasElidableModuleAccessor: Boolean = false
 
-    var fields: List[AnyFieldDef] = Nil
+    var fields: List[AnyFieldDef] = linkedClass.fields
```

Comment:
The previous `setupAfterCreation` did not set `fields`. Is it intentional that, in this refactoring, we now set `fields` anyway? If yes, why? If not, should it stay `Nil` (at least in this commit)? Should it be in the next commit instead?

Reply:
I was concerned about initialization order: In this commit `MethodContainer#updateWith` will try to set `fields` during construction.

Probably I should simply invert this commit and the "update field in walkForChanges". That will be cleaner.

Reply:
That, or initialize it with `= _`, which will make sure that there is no initialization code in this constructor that will override what `updateWith` already set before.